### ISSUE TITLE
Compile with libcurl support

### ISSUE THREAD

**Issue by yeesian** [JuliaGeo/GDALBuilder#2]
To support remote requests

**Comment by visr** [comment]
Yeah so from these two lists there are 15 formats that use libcurl:
- http://www.gdal.org/formats_list.html
- http://www.gdal.org/ogr_formats.html

In the GDAL tarball from the releases page, in logs/GDAL.log, you can see that it is indeed build without cURL support:
```
GDAL is now configured for x86_64-apple-darwin14
  cURL support (wms/wcs/...):no
```

I guess it would be the easiest if someone created a CURLBuilder and switched https://github.com/JuliaWeb/LibCURL.jl to use it?

**Comment by visr** [comment]
Great, it has been done, now we just need to configure it as a dependency. Might be best done in one go with #3.

https://github.com/JuliaWeb/LibCURLBuilder

**Comment by visr** [comment]
While we currently build with `--with-curl=$prefix/bin/curl-config`, it doesn't seem to be working properly, as noted also in the build.log:
```
        found libcurl version 7.64.1
checking for curl_global_init in -lcurl... no
  cURL support (wms/wcs/...):no
```

Too bad, since this would enable many drivers. However I don't want it to hold up https://github.com/JuliaGeo/GDAL.jl/pull/66 any longer.

**Comment by visr** [comment]
Fixed in #12
